monero-rpc/src/wallet.rs

```diff
@@ -0,0 +1,258 @@
+use anyhow::Result;
+use serde::{de::Error, Deserialize, Deserializer, Serialize};
+
+#[jsonrpc_client::api(version = "2.0")]
+pub trait MoneroWalletRpc {
+    async fn get_address(&self, account_index: u32) -> GetAddress;
+    async fn get_balance(&self, account_index: u32) -> GetBalance;
+    async fn create_account(&self, label: String) -> CreateAccount;
+    async fn get_accounts(&self, tag: String) -> GetAccounts;
+    async fn open_wallet(&self, filename: String) -> WalletOpened;
+    async fn close_wallet(&self) -> WalletClosed;
+    async fn create_wallet(&self, filename: String, language: String) -> WalletCreated;
+    async fn transfer(
+        &self,
+        account_index: u32,
+        destinations: Vec<Destination>,
+        get_tx_key: bool,
+    ) -> Transfer;
+    async fn get_height(&self) -> BlockHeight;
+    async fn check_tx_key(&self, txid: String, tx_key: String, address: String) -> CheckTxKey;
+    #[allow(clippy::too_many_arguments)]
+    async fn generate_from_keys(
+        &self,
+        filename: String,
+        address: String,
+        spendkey: String,
+        viewkey: String,
+        restore_height: u32,
+        password: String,
+        autosave_current: bool,
+    ) -> GenerateFromKeys;
+    async fn refresh(&self) -> Refreshed;
+    async fn sweep_all(&self, address: String) -> SweepAll;
+    async fn get_version(&self) -> Version;
+}
+
+#[jsonrpc_client::implement(MoneroWalletRpc)]
+#[derive(Debug, Clone)]
+pub struct Client {
+    inner: reqwest::Client,
+    base_url: reqwest::Url,
+}
+
+impl Client {
+    /// Constructs a monero-wallet-rpc client with localhost endpoint.
+    pub fn localhost(port: u16) -> Self {
+        Client::new(
+            format!("http://127.0.0.1:{}/json_rpc", port)
+                .parse()
+                .expect("url is well formed"),
+        )
+    }
+
+    /// Constructs a monero-wallet-rpc client with `url` endpoint.
+    pub fn new(url: reqwest::Url) -> Self {
+        Self {
+            inner: reqwest::Client::new(),
+            base_url: url,
+        }
+    }
+
+    /// Transfers `amount` monero from `account_index` to `address`.
+    pub async fn transfer_single(
```

Comment:
👍 nice cleanup of the `transfer` usage! (The previously introduced `multi_transfer` was always confusing)